src/query/service/src/interpreters/interpreter_table_modify_column.rs

```diff
@@ -190,22 +190,20 @@ impl ModifyTableColumnInterpreter {
     async fn do_set_data_type(
         &self,
         table: &Arc<dyn Table>,
-        modify_table_schema: &TableSchemaRef,
-        field_comments: &[String],
+        field_and_comments: &[(TableField, String)],
     ) -> Result<PipelineBuildResult> {
         let schema = table.schema().as_ref().clone();
         let table_info = table.get_table_info();
         let mut new_schema = schema.clone();
 
         // first check default expr before lock table
-        for field in modify_table_schema.fields() {
-            let column = &field.name;
-            let default_expr_opt = &field.default_expr;
+        for (field, _comment) in field_and_comments {
+            let column = &field.name.to_string();
             let data_type = &field.data_type;
             if let Ok(i) = schema.index_of(column) {
-                if let Some(default_expr) = default_expr_opt {
-                    new_schema.fields[i].data_type = data_type.clone();
+                if let Some(default_expr) = &field.default_expr {
```

Comment:
As we have analyzed the computed column, we should also set `computed_expr`.
Some tests for this case also need to be added.